File: app/src/main/java/com/creat/motiv/Adapters/MainAdapter.java
package com.creat.motiv.Adapters;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.creat.motiv.Fragments.FavoritesFragment;
import com.creat.motiv.Fragments.HomeFragment;
import com.creat.motiv.Fragments.ProfileFragment;

public class MainAdapter extends FragmentPagerAdapter {
    public MainAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new HomeFragment(); //ChildFragment1 at position 0
            case 1:
                return new FavoritesFragment(); //ChildFragment2 at position 1
            case 2:
                return new ProfileFragment(); //ChildFragment3 at position 2
        }
        return null; //does not happen
    }

    @Override
    public int getCount() {
        return 3; //three fragments
    }
}

File: app/src/main/java/com/creat/motiv/adapters/MainAdapter.kt
package com.creat.motiv.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.creat.motiv.view.fragments.*
class MainAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> HomeFragment()
            1 -> NewQuoteFragment()
            else -> ProfileFragment()
        }


    }

    override fun getCount(): Int {
        return 3//three fragments
    }
}
